src/reducers/historyReducer.js

```diff
@@ -71,34 +83,35 @@ export default function historyReducer(
         ...state,
         data: get(action.payload, 'history.data', {}),
       };
-    case ADD_TRANSACTION:
-      const accountTrxs = state.data[action.payload.accountId] || [];
-      const trxs = {
-        ...state.data,
-        [action.payload.accountId]: [...accountTrxs, action.payload.historyTx],
-      };
+    case ADD_HISTORY_TRANSACTION:
+      ({ accountId } = action.payload);
+      const accountHistory = state.data[accountId];
 
-      return Object.assign(
-        {},
-        state,
-        { data: trxs },
-      );
+      return {
+        ...state,
+        data: {
+          ...state.data,
+          [accountId]: addAccountTransaction(accountHistory, action.payload),
+        },
+      };
     case SET_HISTORY:
-      return Object.assign(
-        {},
-        state,
-        { isFetched: true, data: action.payload, updatingTransaction: null },
-      );
+      return {
+        ...state,
+        data: action.payload,
+        isFetched: true,
+        updatingTransaction: null,
```

Comment:
is `updatingTransaction` a boolean flag or is null expected?

Reply:
It's a `string` if present and `null` if reset.